### PR TITLE
Python Postgres Destination

### PR DESCRIPTION
**Pull request by senyaoh**
Added Python postgres destination

### CODE REVIEW

File: python/destinations/Postgres/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Python Postgres - Destination
+
+Postgres Sink
```
Comment (SteveRosam) [inline python/destinations/Postgres/README.md:3]: Needs more description

File: python/destinations/Postgres/repo.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      "Type": "EnvironmentVariable",
+      "InputType": "InputTopic",
+      "Description": "This is the input topic for demo data",
+      "DefaultValue": "f1-data",
+      "Required": true
+    }
+  ],
+  "EntryPoint": "build/dockerfile",
+  "RunEntryPoint": "main.py"
+}
```
Comment (SteveRosam) [inline python/destinations/Postgres/repo.json:14]: See other review for comments on repo.json

**Review by SteveRosam — APPROVED** [review]

File: python/destinations/Postgres/queue_helper.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import time
+from queue import Queue
+import threading
+from setup_logger import logger
+
+from postgres_helper import create_column, Null, insert_row
+
+def insert_from_queue(conn, table_name: str, insert_queue: Queue, wait_interval: float, batch_size: int):
```
Comment (tomas-quix) [inline python/destinations/Postgres/queue_helper.py:8]: consume_queue as a name for this method is probably easier to follow...

File: python/destinations/Postgres/quix_function.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        self.input_stream = input_stream
+        self.data_start = Null()
+        self.data_end = Null()
+        self.insert_parents()
+        self.topic = os.environ["input"].replace('-', '_')
+        self.committing = False
+
+    def on_committing(self):
+        logger.debug("on_committing")
+        self.committing = True
```
Comment (tomas-quix) [inline python/destinations/Postgres/quix_function.py:23]: This is not going to work. We have to change this.

File: python/destinations/Postgres/quix_function.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
+from quixstreaming import ParameterData, EventData, StreamEndType, StreamReader
+
+import os
+from queue import Queue
+from setup_logger import logger
+from postgres_helper import create_column, insert_row, delete_row, Null
+
+class QuixFunction:
+
+    def __init__(self, conn, table_name, insert_queue: Queue, input_stream: StreamReader):
```
Comment (tomas-quix) [inline python/destinations/Postgres/quix_function.py:10]: Lets keep arguments strongly typed everywhere possible.

File: python/destinations/Postgres/main.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+logger.info("Opening input topic")
+input_topic = client.open_input_topic(os.environ["input"])
+logger.info(os.environ["input"])
+
+# Initialize Queue
+insert_queue = Queue(maxsize=0)
+
+# Create threads that execute insert from Queue
+NUM_THREADS = 10
```
Comment (tomas-quix) [inline python/destinations/Postgres/main.py:51]: Default to 1.

File: python/destinations/Postgres/quix_function.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+        self.data_end = Null()
+        self.insert_parents()
+        self.topic = os.environ["input"].replace('-', '_')
+        self.committing = False
+
+    def on_committing(self):
+        logger.debug("on_committing")
+        self.committing = True
+
+    def reset_data_ts(self):
```
Comment (tomas-quix) [inline python/destinations/Postgres/quix_function.py:25]: We don't need this.

File: python/destinations/Postgres/quix_function.py [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+        row = {'timestamp': data.timestamp_nanoseconds}
+
+        for k, v in data.tags.items():
+            create_column(
+                self.conn, self.table_name["EVENT_TABLE_NAME"], 'TAG_' + k, 'STRING')
+            row['TAG_' + k] = v
+
+        row['value'] = data.value
+        insert_row(self.conn, self.table_name["EVENT_TABLE_NAME"], list(
```
[the remark this answers is not in the document]
Comment (tomas-quix) [reply python/destinations/Postgres/quix_function.py:100]: or just use same one as you want.

**Pull request merged**